### PR TITLE
Add eflow VM in CI

### PR DESCRIPTION
**Pull request by Mandur**
# PR for issue #692

# I could not yet experiment with PR trigger and I will do it in this in a follow up PR

## What is being addressed

We want to test EFLOW in the nightly test 

## How is this addressed

* CI is changed to target environment which could be swapped in an easy manner
* Migration to the new Azure Subscription LoRa E2E tests
* A VM on Azure run EFLOW on a gateway, a local concentrator forward message there
* A VPN permit the Azure VM to seed back the logs on the local test runner.
* CI is using some custom composite action to avoid code repetition
* E2E tests are now running in the same job to avoid the 3 minutes lead build and setup time
* CI run only on week days

## Future Work
* [PR trigger](#810)
* [Improved security with compartmentalized rights token for ACR access](#811 )
* [Turn on and off by the Azure VM from the CI](#812 )
* [Remove completely the Azure iot edge toolkit used for token replacement](#813 )

### CODE REVIEW

File: LoRaEngine/deployment.ci.aio.template.json [hunk 1]
```diff
@@ -50,16 +50,10 @@
             },
             "env": {
               "OptimizeForPerformance": {
-                "value": "$EDGEHUB_OPTIMIZEFORPERFORMANCE"
+                "value": "false"
```
Comment (Mandur) [inline LoRaEngine/deployment.ci.aio.template.json:53]: needs to be variable

File: LoRaEngine/deployment.ci.aio.template.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
             "status": "running",
             "restartPolicy": "always",
             "settings": {
-              "image": "mcr.microsoft.com/azureiotedge-hub:$EDGE_HUB_VERSION",
+              "image": "mcr.microsoft.com/azureiotedge-hub:1.2.2",
```
Comment (Mandur) [inline LoRaEngine/deployment.ci.aio.template.json:33]: needs to be variable

File: .github/actions/deployedge/action.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
+name: 'Deploy IoT Edge'
+description: 'Greet someone'
```
Comment (danigian) [inline .github/actions/deployedge/action.yml:2]: Insert a valid description

File: .github/actions/iotedgedev/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+sudo -E iotedgedev genconfig -f $2/$1 -P $3  --fail-on-validation-error
```
Comment (danigian) [inline .github/actions/iotedgedev/entrypoint.sh:3]: Can we add a description of what this file does and why, at the beginning of the file itself?

File: .github/workflows/e2e-ci.yaml [hunk 5]
```diff
@@ -2,23 +2,42 @@ name: LoRa E2E CI
...
+  # these environments variable are for the deployment to IoT Edge
+  CONTAINER_REGISTRY_ADDRESS: "${{ secrets.CI_ACR_NAME }}.azurecr.io"
+  CONTAINER_REGISTRY_USERNAME: "${{ secrets.CI_ACR_NAME }}"
+  CONTAINER_REGISTRY_PASSWORD: ${{ secrets.CI_ACR_CREDENTIALS }}
+  IOT_EDGE_DEPLOYMENT_ID: ""
+  NET_SRV_LOG_LEVEL: 1
+  NET_SRV_LOGTO_HUB: false
+  NET_SRV_LOGTO_UDP: true
+  NET_SRV_IOTEDGE_TIMEOUT: 0
+  NET_SRV_LOG_TO_UDP_ADDRESS: "itestup"
```
Comment (danigian) [inline .github/workflows/e2e-ci.yaml:30]: Should this be fixed?

File: .github/workflows/e2e-ci.yaml [hunk 6]
```diff
@@ -124,204 +148,244 @@ jobs:
...
+          architecture: '${{ secrets.ARCHITECTURE }}'
+          iot_edge_deployment_id: '${{ secrets.IOT_EDGE_DEPLOYMENT_ID }}'
 
+  # Deploy IoT Edge solution to standalone Concentrator
+  deploy_eflow_concentrator_iot_edge:
+    needs:
+    - env_var
+    - build_push_docker_images
+    environment:
+      name: CI_CONCENTRATOR_EFLOW_ARM32
```
Comment (danigian) [inline .github/workflows/e2e-ci.yaml:248]: Isn't this name misleading? It's not EFLOW_ARM32, it's a RPi no?

Comment (Mandur) [reply .github/workflows/e2e-ci.yaml:266]: It's a concentrator running on arm32 used to target messages to eflow. This is why the name. I could change it if you don't like it

File: .github/workflows/e2e-ci.yaml (hunk 5 above)
Comment (Mandur) [reply .github/workflows/e2e-ci.yaml:30]: Is this comment in the scope of the migration to TCP? I am probably going to merge first as Atif wants to run a full ci on hist branch @atifaziz

File: .github/workflows/e2e-ci.yaml (hunk 6 above)
Comment (danigian) [reply .github/workflows/e2e-ci.yaml:266]: It's fine, I misread 👍

**Review by danigian — APPROVED** [review]

**Pull request merged**